src/operations.jl

```diff
@@ -1,91 +1,139 @@
 import Base.+, Base.-, Base.*, LinearAlgebra.mul!
 
+function mul!(res::AbstractVector, op::AbstractLinearOperator{T}, v::AbstractVector, α, β) where T 
+  (size(v, 1) == size(op, 2) && size(res, 1) == size(op, 1)) || throw(LinearOperatorException("shape mismatch"))
+  increase_nprod(op)
+  op.prod!(res, v, α, β)
+end
+
+function mul!(res::AbstractVector, op::AbstractLinearOperator{T}, v::AbstractVector) where T
+  mul!(res, op, v, one(T), zero(T))
+end
+
 # Apply an operator to a vector.
 function *(op::AbstractLinearOperator{T}, v::AbstractVector{S}) where {T, S}
-  size(v, 1) == size(op, 2) || throw(LinearOperatorException("shape mismatch"))
-  increase_nprod(op)
-  op.prod(v)::typeof(v).name.wrapper{promote_type(T, S), typeof(v).parameters[2]}
+  nrow, ncol = size(op)
+  res = Vector{promote_type(T, S)}(undef, nrow)
```

Comment:
It will not work with GPU arrays.

Reply:
That line actually caused it to not work properly on cpu...

Reply:
Would `similar(v, promote_type(T, S), nrow)` solve the issue with GPU arrays?

Reply:
Yes! I just tested it with CuVectors.